include/seqan/modifier/modifier_iterator.h

```diff
@@ -130,14 +130,8 @@ struct Value<ModifiedIterator<THost, TSpec> const> : Value<ModifiedIterator<THos
 // Metafunction GetValue
 // --------------------------------------------------------------------------
 
-//NOTE(h-2): ModifiedStringIterators always return by value since some
-// modified strings result in values being created anyway (e.g. ModViews)
-// and depending on scope these might be lost.
-// For example iterators over infixes of ModViews would otherwise reference
-// part of the stack that were freed.
-
 template <typename THost, typename TSpec>
-struct GetValue< ModifiedIterator<THost, TSpec> > : Value<THost>
+struct GetValue< ModifiedIterator<THost, TSpec> > : GetValue<THost>
```

Comment:
No, I have moved the special logic to modview so that it doesn't effect all mod strings, but only the modviews.